src/main/cpp/subsystems/DriveSubsystem.cpp

```diff
@@ -0,0 +1,207 @@
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+#include "subsystems/DriveSubsystem.h"
+
+#include <frc/geometry/Rotation2d.h>
+#include <frc/kinematics/DifferentialDriveWheelSpeeds.h>
+#include <frc/smartdashboard/SmartDashboard.h>
+#include <frc/trajectory/constraint/DifferentialDriveVoltageConstraint.h>
+
+#include "Constants.h"
+
+using namespace DriveConstants;
+
+DriveSubsystem::DriveSubsystem() {
+    // Implementation of subsystem constructor goes here.
+    // Stuff you want to happen once, when robot code starts running
+
+    // Initialize each motor with MotionMagic settings
+    // Made this a function since we do the same thing four times
+    ConfigureMotor(&RightLead);
+    ConfigureMotor(&RightFollow);
+    ConfigureMotor(&LeftLead);
+    ConfigureMotor(&LeftFollow);
+
+    // Invert left side, since DifferentialDrive no longer does it for us
+    LeftLead.SetInverted(true);
+    LeftFollow.SetInverted(true);
+
+    // Drive train motor grouping start
+    RightFollow.Follow(RightLead);
+    LeftFollow.Follow(LeftLead);
+
+    // Start encoders at zero
+    ResetEncoders();
+
+    // setup for trajectories
+    trajectoryConfig = new frc::TrajectoryConfig(kMaxSpeed, kMaxAcceleration);
+    // Create a voltage constraint to ensure we don't accelerate too fast
+    frc::DifferentialDriveVoltageConstraint autoVoltageConstraint(
+        frc::SimpleMotorFeedforward<units::meters>(
+            DriveConstants::ks, DriveConstants::kv, DriveConstants::ka),
+        DriveConstants::kDriveKinematics, 10_V);
+
+    // Add kinematics to ensure max speed is actually obeyed
+    trajectoryConfig->SetKinematics(DriveConstants::kDriveKinematics);
+    // Apply the voltage constraint
+    trajectoryConfig->AddConstraint(autoVoltageConstraint);
+
+    frc::SmartDashboard::PutNumber("KpDistance", -2);
+}
+
+void DriveSubsystem::DisabledInit() {}
+
+void DriveSubsystem::TeleopInit() {
+    ConfigureMotor(&RightLead);
+    ConfigureMotor(&RightFollow);
+    ConfigureMotor(&LeftLead);
+    ConfigureMotor(&LeftFollow);
+}
+
+void DriveSubsystem::SetCoast(WPI_TalonFX *_talon) {
+    _talon->SetNeutralMode(Coast);
+}
+
+void DriveSubsystem::Periodic() {
+    // Implementation of subsystem periodic method goes here.
+    // Things that happen while robot is running */
+
+    m_odometry.Update(currentrobotAngle,
+                      units::meter_t(lEncoder * kEncoderDistancePerPulse),
+                      units::meter_t(rEncoder * kEncoderDistancePerPulse));
+
+    // Do things when first enabled or disabled
+    if (frc::DriverStation::IsDisabled() != true &&
```

Comment:
Don't compare to `true`/`false`. Just do `!frc::DriverStation::IsDisabled()` instead